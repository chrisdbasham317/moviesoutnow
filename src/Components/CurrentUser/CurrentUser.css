.currentUser__div {
  display: flex;
  align-items: center;
}

.currentUser__hidden {
  display: none;
}

.currentUser__p--name{
  color: #FFF;
  margin-right: 20px;
  font-size: 20px;
}

.view--favorite--movies__button{
  color: #FFF;
  font-size: 20px;
  margin-right: 20px;
  background: none;
  border: none;
  cursor: pointer;
  color: #FDC938;
}

.sign--out__button {
  border: none;
  height: 30px;
  width: 80px;
  border-radius: 5px;
  background-color: #F541FF;
  color: #FFF;
  font-weight: 700;
  font-size: 15px;
  cursor: pointer;
  text-decoration: none;
  
}