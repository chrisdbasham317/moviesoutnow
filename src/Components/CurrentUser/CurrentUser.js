import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchFavorites } from '../../Thunks/fetchFavorites';
import './CurrentUser.css';
import { signOut } from '../../Actions';
import { setFavorites } from '../../Actions';

export class CurrentUser extends React.Component{

  handleClick = (event) => {
    event.preventDefault();
    this.props.signOut();
    this.props.setFavorites([]);
  }

  render() {
    const userId = this.props.currentUser.id
    return (
      <div className={this.props.currentUser.id ? 'currentUser__div' : 'currentUser__hidden'}>
        <p className="currentUser__p--name">Welcome, {this.props.currentUser.name}!</p>
        <Link to='/'>
          <button
            type="button"
            className="go-home__button">Home
          </button>
        </Link>
        <Link to='/favorites'>
          <button
            type="button"
            className="view--favorite--movies__button">Favorite Movies
          </button>
        </Link>
        <button
            type="button"
            className="sign--out__button"
            onClick={(event) => this.handleClick(event)}>
          <Link to='/' className='sign--out__button'>
            Sign Out
          </Link>
        </button>
      </div>
    )
  }
}

const mapStateToProps = ({ currentUser, favorites }) => ({
  currentUser,
  favorites
})

const mapDispatchToProps = (dispatch) => (
  bindActionCreators({ signOut, fetchFavorites, setFavorites }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(CurrentUser)