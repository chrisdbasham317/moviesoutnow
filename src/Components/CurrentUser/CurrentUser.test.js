import { CurrentUser, mapStateToProps, mapDispatchToProps } from './CurrentUser';


describe('CurrentUser', () => {
    it('should match the snapshot with all data passed in correctly', () => {

    })
})