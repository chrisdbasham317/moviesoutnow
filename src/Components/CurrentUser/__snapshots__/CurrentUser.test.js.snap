// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentUser should match the snapshot with all data passed in correctly 1`] = `
<div
  className="currentUser__div"
>
  <p
    className="currentUser__p--name"
  >
    Welcome, 
    Chris
    !
  </p>
  <Link
    to="/favorites"
  >
    <button
      className="view--favorite--movies__button"
      type="button"
    >
      Favorite Movies
    </button>
  </Link>
  <button
    className="sign--out__button"
    onClick={[Function]}
    type="button"
  >
    <Link
      className="sign--out__button"
      to="/"
    >
      Sign Out
    </Link>
  </button>
</div>
`;
